js/listanchors.js

```diff
@@ -88,6 +88,65 @@ function listanchors()
         id += key;
         var e = document.getElementById(id);
         e.innerHTML = newText;
+
+    }
+    function addVersionSelector() {
+      // Latest version offered by the archive builds
+      // This needs to be manually updated after new versions have been archived
+      var currentArchivedVersion = 74;
+      // build URLs for dlang.org: DDoc + Dox
+      var ddocModuleURL = document.body.id.replace(/[.]/g, "_") + ".html";
+      var ddoxModuleURL = document.body.id.replace(/[.]/g, "/") + ".html";
+
+      // build list of versions available in the docarchives
+      var archivedVersions = [];
+      while (currentArchivedVersion >= 66) {
+        archivedVersions.push("2.0" + currentArchivedVersion--);
+      }
+      archivedVersions = archivedVersions.map(function(e) {
+          return {
+            name: e,
+            url: "https://docarchives.dlang.io/v" + e + ".0/phobos/" + ddocModuleURL,
+            selected: false,
+          };
+      });
+
+      var rootURL = location.href.split(/\/(phobos|library)(-prerelease)?/)[0]
+      var onlineVersions = [{
+        name: "master",
+        url: rootURL + "/phobos-prerelease/" + ddocModuleURL,
+      },{
+        name: "master (beta)",
+        url: rootURL + "/library-prerelease/" + ddoxModuleURL,
+      },{
+        name: "stable",
+        url: rootURL + "/phobos/" + ddocModuleURL,
+      },{
+        name: "stable (beta)",
+        url: rootURL + "/library/" + ddoxModuleURL,
+      }];
+
+      // set the current URL as selected
+      var currentURL = location.href.split(/[#?]/)[0];
+      onlineVersions.forEach(function(v, i) {
+        onlineVersions[i].selected = v.url === currentURL;
+      });
+
+      // build select box of all versions and append to current DOM
+      var versions = onlineVersions.concat(archivedVersions);
+      var options = versions.map(function(e, i){
+        return "<option value='" + i + "'" + (e.selected ? "selected" : "") + ">" + e.name + "</option>";
+      });
+      $("h1").after("<div class='version-changer-container fa-select'><select id='version-changer'>" + options.join("") + "</select></div>");
```

Comment:
> I'm pretty sure you're supposed to put the element you're floating before the element it is supposed to "overlap", 

I tried `before` as well, but discarded the approach due as floating didn't look nice for mobile devices. Though of course, now I already added viewport-specific CSS.
For reference, currently it looks like this:

![image](https://user-images.githubusercontent.com/4370550/27942899-23b562c0-62db-11e7-8db8-8e6d1e8ef820.png)

With `before` it could only look like this:

![image](https://user-images.githubusercontent.com/4370550/27942893-1b13dd22-62db-11e7-9ff3-54dd2c120539.png)

Though of course, we could simple hide the select box on mobile devices.

> Why does this work for you?

```css
.version-changer-container {
    margin-top: -4em;
}
```

The interesting question is why it doesn't work for ;-)
A couple of external tools show the same results:

https://tools.pingdom.com/#!/bpOIFJ/http://dtest.dlang.io/artifact/website-b29c592343b3eca4d1a2bb7774123c738e67b788-7991f163dc4768833cc161ca4595703d/web/phobos-prerelease/core_bitop.html
https://www.webpagetest.org/result/170707_0E_8F6/

Google even allows to preview the mobile viewport (this one has special CSS - no floating):

https://developers.google.com/speed/pagespeed/insights/?url=http%3A%2F%2Fdtest.dlang.io%2Fartifact%2Fwebsite-b29c592343b3eca4d1a2bb7774123c738e67b788-7991f163dc4768833cc161ca4595703d%2Fweb%2Fphobos-prerelease%2Fcore_bitop.html